src/main/java/com/puppycrawl/tools/checkstyle/checks/metrics/NPathComplexityCheck.java

```diff
@@ -51,18 +51,28 @@
     /** The initial current value. */
     private static final BigInteger INITIAL_VALUE = BigInteger.ONE;
 
-    /** Stack of values - all but the current value. */
-    private final Deque<BigInteger> valueStack = new ArrayDeque<>();
+    /**
+     * Stack of NP values for ranges.
+     */
+    private final Deque<BigInteger> rangeValues = new ArrayDeque<>();
+
+    /** Stack of NP values for expressions. */
+    private final Deque<Integer> expressionValues = new ArrayDeque<>();
```

Comment:
ok, npath above 2,147,483,647 (Integer limit) is smth really strange. And does not mean any useful metric. BigInterger will work always but it is performance hit.
Ok, lets keep it as is for now.